azure-pipelines.yaml

```diff
@@ -3,12 +3,97 @@
-- task: Bash@3
-  displayName: "Set build number"
-  inputs:
-    targetType: inline
-    script: 'echo hello world'
 
+stages:
+  - stage: CI
+    jobs:
+      - job: Validate
+        pool:
+          vmImage: 'Ubuntu-16.04'
+        steps:
+          - task: PublishPipelineArtifact@0
+            inputs:
+              artifactName: 'drop'
+              targetPath: '$(System.DefaultWorkingDirectory)/template.json'
+  - stage: Sandbox
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    jobs:
+      - deployment: Deploy
+        pool:
+          vmImage: 'Ubuntu-16.04'
+        environment: 'sandbox'
+        strategy:
+          # default deployment strategy
+          runOnce:
+            deploy:
+              steps:
+                - task: DownloadPipelineArtifact@1
+                  inputs:
+                    buildType: 'current'
+                    artifactName: 'drop'
+                    targetPath: '$(System.DefaultWorkingDirectory)'
+                - task: AzureResourceGroupDeployment@2
+                  displayName: "Deploy hmctssandbox"
+                  inputs:
+                    azureSubscription: 'azurerm-sandbox'
+                    action: 'Create Or Update Resource Group'
+                    resourceGroupName: 'cnp-acr-rg'
+                    location: 'UK South'
+                    deploymentMode: 'Incremental'
+                    templateLocation: 'Linked artifact'
+                    csmFile: 'template.json'
+                    overrideParameters: '-name hmctssandbox'
+  - stage: Prod
+    condition: and(succeeded(), eq(variables['Build.SourceBranch'], 'refs/heads/master'))
+    dependsOn: Sandbox
+    jobs:
+      - deployment: hmctspublic
+        pool:
+          vmImage: 'Ubuntu-16.04'
+        environment: 'prod'
+        strategy:
+          # default deployment strategy
+          runOnce:
+            deploy:
+              steps:
+                - task: DownloadPipelineArtifact@1
+                  inputs:
+                    buildType: 'current'
+                    artifactName: 'drop'
+                    targetPath: '$(System.DefaultWorkingDirectory)'
+                - task: AzureResourceGroupDeployment@2
+                  displayName: "Deploy hmctspublic"
+                  inputs:
+                    azureSubscription: 'azurerm-prod'
+                    action: 'Create Or Update Resource Group'
+                    resourceGroupName: 'rpe-acr-prod-rg'
+                    location: 'UK South'
+                    deploymentMode: 'Incremental'
+                    templateLocation: 'Linked artifact'
+                    csmFile: 'template.json'
+                    overrideParameters: '-name hmctspublic'
+      - deployment: hmctsprivate
+        pool:
+          vmImage: 'Ubuntu-16.04'
+        environment: 'prod'
+        strategy:
+          # default deployment strategy
+          runOnce:
+            deploy:
+              steps:
```

Comment:
May be create a new step template and reuse ? But this might never know, may not be worth it .

Reply:
cbf, this will likely never even run again. waste of time even building this imo 😉